python/ray/data/dataset.py

```diff
@@ -2619,6 +2625,21 @@ def write_datasource(
         finally:
             progress.close()
 
+    def iterator(self) -> DatasetIterator:
+        """Return a :class:`DatasetIterator <ray.data.DatasetIterator>` that
```

Comment:
Nit: Ref link.
```suggestion
        """Return a :class:`~ray.data.DatasetIterator` that
```

Reply:
Also, the `ray.data.DatasetIterator` currently doesn't work in the docs, not sure why. 🤔

Reply:
Hmm will look into it.